Tests/test_SeqIO.py

```diff
@@ -4386,16 +4386,11 @@ def test_snapgene1(self):
     def test_empty_file(self):
         """Check parsers can cope with an empty file."""
         for t_format in SeqIO._FormatToIterator:
-            if t_format in SeqIO._BinaryFormats:
-                handle = BytesIO()
-                with self.assertRaisesRegexp(ValueError, "Empty file."):
-                    list(SeqIO.parse(handle, t_format))
-            elif t_format in ("uniprot-xml", "pdb-seqres", "pdb-atom", "cif-atom", "cif-seqres"):
-                handle = StringIO()
+            handle = StringIO()
+            if t_format in ("abi", "abi-trim", "cif-atom", "cif-seqres", "gck", "nib", "pdb-atom", "pdb-seqres", "seqxml", "sff", "sff-trim", "snapgene", "uniprot-xml", "xdna"):
```

Comment:
Does this check for binary files work even with ``StringIO`` instead of ``BytesIO`` on the assumption that the parsers check for empty handles before any check on the mode?

Reply:
Yes it works. In any case, because Python uses duck-typing, parsers should not check the mode.